### DISCUSSION TITLE
Could not find a declaration file for module sortablejs and dompurify

### DISCUSSION THREAD

**Discussion by Raw-Gel** [ghiscoding/Angular-Slickgrid#1166] [Q&A] [answered]
Hi I'm new in using the slick grid and I follow the HOWTO Step by Step procedure but when I start the project it return an error like this  
![354043725_151034341315467_7672345406523851562_n](https://github.com/ghiscoding/Angular-Slickgrid/assets/52986376/2d4edfbd-86d4-4259-8480-2cd235073d1e)

**Comment by ghiscoding** [comment]
which version is that with? Can you try to clone the [Angular-Slickgrid-Demos](https://github.com/ghiscoding/angular-slickgrid-demos) instead, it works correctly, also maybe check that you're using `"allowSyntheticDefaultImports": true,` in your `tsconfig.json` file...or actually you might be missing the `typesRoot` in your tsconfig, both DOMPurify and SortableJS have their types defined in `node_modules/@types` so your tsconfig should have this setting to get them

https://github.com/ghiscoding/Angular-Slickgrid/blob/994cfe715efcda49a1d812ac7dbe61db3f1a62da/tsconfig.json#L19C4-L22

```ts
// tsconfig.json and/or tsconfig.app.json
"typeRoots": [
      "node_modules/@types"
],
```

**Reply by ghiscoding** [reply, marked as answer]
I thought about it a bit more and it might be because these Types are not required (`devDependencies` in Slickgrid-Universal) and for that reason it might be why you're not resolving the types. I would guess that the 2 following would probably work

1. add SortableJS and DOMPurify types to your `devDependencies`
```shell
npm install -D @types/sortablejs
npm install -D @types/dompurify
```
2. modify your `tsconfig.json` to ignore types from external libs
```js
// tsconfig.json or tsconfig.app.json
{
  "compilerOptions": {
    "skipLibCheck": true
  }
}
```

EDIT

I added `@types/sortablejs` as a dependency into Slickgrid-Universal so this error will never show up anymore. On the other hand, the `@types/dompurify` was not added to the dependency list because DOMPurify was made optional in latest major version, so if you install DOMPurify (recommended) then you will want to install its Type at the same time.

So I consider this problem resolved with info provided above
